python/tvm/autotvm/record.py

```diff
@@ -258,7 +267,7 @@ def pick_best(in_file, out_file):
     logger.info("Extract %d best records from the %s", len(best_set), in_file)
     fout = open(out_file, 'w') if isinstance(out_file, str) else out_file
 
-    for inp, res in load_from_file(in_file):
+    for inp, res in context_clone:
```

Comment:
Why do we need a clone?

Reply:
Generators can only be iterated through once and ApplyBestHistory uses up context. We need a fresh generator during the actual write. We either have to use a tee clone or recreate the generator. I thought this way was more concise.